plugins/transformations/json/srcC/FlatTypeTransformation.c

```diff
@@ -1150,165 +635,179 @@ DDS_ReturnCode_t RTI_TSFM_Json_FlatTypeTransformation_serialize(
-
-    /* json_buffer should not be printed with printf() because there is
-     * no guarantee that the string is well terminated (and most likely
-     * it won't be, i.e. no 'nul' terminator) */
-
-    value = json_parse(json_buffer, json_buffer_size);
-    if (value == NULL) {
-        /* TODO Log error */
-        RTI_TSFM_ERROR("failed: json_parse")
-        goto done;
+    /* The EOS character cannot be added without allocating new memory */
+    if (max == length) {
+        return DDS_BOOLEAN_FALSE;
     }
-    if (value->type != json_object) {
-        /* TODO Log error */
-        RTI_TSFM_ERROR("failed: invalid parsed value")
-        goto done;
-    }
-
-    mappings_len =
-            RTI_TSFM_Json_FlatTypeTransformation_MemberMappingSeq_get_length(
-                    &self->state->output_mappings);
-
-    for (i = 0; i < mappings_len; i++) {
-        mapping =
-                RTI_TSFM_Json_FlatTypeTransformation_MemberMappingSeq_get_reference(
-                        &self->state->output_mappings,
-                        i);
-        if (DDS_RETCODE_OK
-            != RTI_TSFM_Json_FlatTypeTransformation_find_json_member(
-                    mapping->name,
-                    value,
-                    &member)) {
-            /* TODO Log error */
-            goto done;
-        }
-        if (member == NULL) {
-            if (!mapping->optional) {
-                RTI_TSFM_ERROR_1(
-                        "required member not found in JSON sample:",
-                        "member=%s",
-                        mapping->name)
-                goto done;
-            }
-            continue;
-        }
 
-        if (DDS_RETCODE_OK
-            != RTI_TSFM_Json_FlatTypeTransformation_parse_member(
-                    self,
-                    mapping,
-                    member->value,
-                    sample)) {
-            /* TODO Log error */
-            goto done;
-        }
+    /*
+     * In this case, we have at least one 'emtpy' space in the sequence,
+     * therefore, we can add the EOS character.
+     */
+    if (!DDS_OctetSeq_set_length(self, length + 1)) {
+        return DDS_BOOLEAN_FALSE;
     }
 
-#if 0
-    DDS_DynamicData_print(sample, stdout, 0);
-#endif
+    /*
+     * length contains the length before increasing it,
+     * hence it is pointing ot the new element.
+     */
+    *DDS_OctetSeq_get_reference(self, length) = '\0';
 
-    retcode = DDS_RETCODE_OK;
-
-done:
-    if (value != NULL) {
-        json_value_free(value);
-    }
+    return DDS_BOOLEAN_TRUE;
 
-    return retcode;
 }
 
-
 DDS_ReturnCode_t RTI_TSFM_Json_FlatTypeTransformation_deserialize(
         RTI_TSFM_UserTypePlugin *plugin,
         RTI_TSFM_Transformation *transform,
         DDS_DynamicData *sample_in,
         DDS_DynamicData *sample_out)
 {
-    DDS_ReturnCode_t retcode = DDS_RETCODE_ERROR;
+    DDS_ReturnCode_t retcode = DDS_RETCODE_OK;
```

Comment:
I saw a couple of conditions in which the retcode was not modified. I tried to reuse the retcode that some functions return so we can get more info about the issue (out of resources, precondition not met...). In this case, before all `goto done` we should modify the retcode (either manually or from an API call), but we get more info in case of error.

Reply:
You can still cache the result to another variable, and use it to set `retcode` inside the error block.